File: Java/src/main/java/com/DSA/Java/Queue/LinkedListQueue/LinkedListQueue.java
package com.DSA.Java.Queue.LinkedListQueue;

public class LinkedListQueue {
   public SinglyLinkedList sll;

   public LinkedListQueue() {
      sll = new SinglyLinkedList();
   }

   public boolean isEmpty() {
      if (sll.head == null) {
         return true;
      }
      return false;
   }

   public void enQueue(int value) {
      sll.add(value, sll.size + 1);
   }

   public int deQueue() {
      if (isEmpty()) {
         System.out.println("The Queue is already empty"); 
         return -1;
      }
      int result = sll.head.value;
      sll.remove(0);
      return result;
   }

   public int peek() {
      if (isEmpty()) {
         System.out.println("The Queue is already empty"); 
         return -1;
      }
      return sll.head.value;
   }

   public void delete() {
      sll.head = sll.tail = null;
      sll.size = 0;
   }
}

File: C++/Queue/src/LinkedListQueue.cpp
#include "../include/LinkedListQueue.h"

bool LinkedListQueue::is_empty() const {
    return sll.head == nullptr;
}

int LinkedListQueue::peek() const {
    if (is_empty()) {
        std::cout << "The Queue is empty" << std::endl;
        return -1;
    }
    return sll.head->value;
}

void LinkedListQueue::en_queue(int value) {
    sll.add_element(value, sll.size + 1);
}

int LinkedListQueue::de_queue() {
    if (is_empty()) {
        std::cout << "The Queue is already empty" << std::endl;
        return -1;
    }
    int result = sll.head->value;
    sll.delete_element(result);
    return result;
}

void LinkedListQueue::delete_queue() {
    sll.delete_ll();
}
